enumerables.rb

```diff
@@ -1,41 +1,151 @@
-      i+=1
+    a = Array self
+    while i < size
+      yield(a[i], i) if block_given?
+      new_array << a[i] unless block_given?
+      i += 1
+    end
+    return Enumerator.new(new_array) unless block_given?
+
+    self
   end
-  array
-end
 
-def my_select
-    return to_enum(:my_select) unless block_given?
+  def my_select
+    return my_each unless block_given?
+
     new_array = []
-  my_each do |x|
-new_array.push(x) if yield(x) == true
+    my_each { |n| new_array.push(n) if yield(n) }
+    new_array
+  end
+
+  def my_all?(*args)
+    result = true
+    if block_given?
+      my_each do |i|
+        result = false unless yield(i)
+      end
+    elsif !args[0].nil?
+      my_each do |i|
+        result = false unless args[0] === i
+      end
+    elsif empty? && args[0].nil?
+      result = true
+    else
+      my_each do |i|
+        result = false unless i
+      end
+    end
+    result
+  end
+
+  def my_any?(*args)
+    result = false
+    if block_given?
+      my_each do |i|
+        result = true if yield(i)
+      end
+    elsif !args[0].nil?
+      my_each do |i|
+        result = true if args[0] === i
+      end
+    elsif empty? && args[0].nil?
+      result = false
+    else
+      my_each do |i|
+        result = true if i
+      end
+    end
+    result
+  end
+
+  def my_none?(args = nil)
+    result = if block_given?
+               !my_any? { |i| yield i }
+             else
+               !my_any?(args)
+             end
+    result
+  end
+
+  def my_count(item = false)
+    count = 0
+
+    if !block_given?
+      return size unless item
+
+      my_each { |n| count += 1 if n == item }
+    elsif block_given? && !item
+      my_each { |n| count += 1 if yield(n) }
+    elsif block_given? && item
+      my_each { |n| count += 1 if n == item }
+    end
+    count
   end
-  new_array
-end
-end
-  
 
-puts [1,2,3,4,5,6].my_select(&:even?)
+  def my_map(args_proc = nil)
+    return to_enum :my_map unless block_given? || args_proc.class == Proc
+    new_arr = []
+    new_object = to_a
+    if !args_proc.nil?
+      new_object.my_each do |i|
+        new_arr.push(proc.call(i))
+      end
+    else
+      new_object.my_each do |i|
+        new_arr.push(yield(i))
+      end
+    end
+    new_arr
+  end
 
+  def my_inject(args1 = nil, args2 = nil)
+    new_object = to_a
+    sum = new_object[0]
+    t = 1
+    if !args1.nil? && args2.class == Symbol
+      sum = args1
+      new_object.my_each { |i| sum = sum.send(args2, i) }
+    elsif args1.nil? && args2.nil?
+      loop do
+        sum = yield(sum, new_object[t]) if block_given?
+        t += 1
+        break if t == new_object.length
+      end
+    elsif args1.class == Symbol
+      loop do
+        sum = sum.send(args1, new_object[t])
+        t += 1
+        break if t == new_object.length
+      end
 
+    elsif !args1.nil? && args2.nil?
+      sum = args1
+      new_object.my_each { |i| sum = yield(sum, i) } if block_given?
+    end
+    sum
+  end
```

Comment:
- [ ] should have working `my_inject` identical and returns the same thing as ruby's [`inject`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-inject). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_inject.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

- raises a "LocalJumpError" when no block or argument is given
```ruby
[1, 5, 8, 10].my_inject == [1, 5, 8, 10].inject 
```
Yours (incorrect)
![image](https://user-images.githubusercontent.com/52209067/84580526-e65f6d00-ad94-11ea-8a62-eb8621ba4208.png)

Real inject (correct)
![image](https://user-images.githubusercontent.com/52209067/84206610-bcdcd380-aa6c-11ea-9186-f9111c66d315.png)